src/components/plots/gridPlot.ts

```diff
@@ -43,28 +48,49 @@ export module Plot {
     }
 
     /**
-     * @param {string} attrToSet One of ["x", "y", "fill"]. If "fill" is used,
+     * @param {string} attrToSet One of ["x", "y", "x2", "y2", "fill"]. If "fill" is used,
      * the data should return a valid CSS color.
      */
     public project(attrToSet: string, accessor: any, scale?: Scale.AbstractScale<any, any>) {
+      attrToSet = attrToSet === "x1" ? "x" : attrToSet;
+      attrToSet = attrToSet === "y1" ? "y" : attrToSet;
       super.project(attrToSet, accessor, scale);
+
+      if (attrToSet === "x" && this._projections["x2"] === undefined) {
```

Comment:
As per our previous discussion, I don't think we need this x2 projection branch anymore.

Something along the lines of

if (attrToSet is x and the scale is Category instanced)
  project onto x1/x2